include/aspect/initial_conditions/interface.h

```diff
@@ -93,6 +94,31 @@ namespace aspect
     };
 
 
+    /**
+     * Create a class inheriting from dealii.Function to translate the
+     * interface call to dealii.Function format
+     */
+    template <int dim>
+    class InterfaceFunction : public dealii::Function<dim, double>
+    {
+      public:
+        InterfaceFunction (const Interface<dim> &interface,
+                           const unsigned int selected_component,
+                           const unsigned int n_components);
```

Comment:
The same applies to the other class, by the way. I think we can also do without it.